MySQL window tests failing

There are some mysql window tests that are failing:

```
FAILED ibis/tests/all/test_window.py::test_window[MySQL-cumnotany] - Attribut...
FAILED ibis/tests/all/test_window.py::test_window[MySQL-cumnotall] - Attribut...
```

the error is:

```
Neither 'BinaryExpression' object nor 'Comparator' object has an attribute 'over'
```

ref #2226

--- a/ibis/tests/all/test_window.py
+++ b/ibis/tests/all/test_window.py
@@ -6,6 +6,7 @@ import ibis.common.exceptions as com
 from ibis.tests.backends import (
     Csv,
     Impala,
+    MySQL,
     OmniSciDB,
     Pandas,
     Parquet,
@@ -111,7 +112,7 @@ from ibis.tests.backends import (
         ),
         param(
             # notany() over window not supported in Impala, PostgreSQL,
-            # and Spark backends
+            # Spark and MySQL backends
             lambda t, win: (t.double_col == 0).notany().over(win),
             lambda t: (
                 t.double_col.expanding()
@@ -120,7 +121,9 @@ from ibis.tests.backends import (
                 .astype(bool)
             ),
             id='cumnotany',
-            marks=pytest.mark.xfail_backends((Impala, PostgreSQL, Spark)),
+            marks=pytest.mark.xfail_backends(
+                (Impala, PostgreSQL, Spark, MySQL)
+            ),
         ),
         param(
             lambda t, win: (t.double_col == 0).all().over(win),
@@ -134,7 +137,7 @@ from ibis.tests.backends import (
         ),
         param(
             # notall() over window not supported in Impala, PostgreSQL,
-            # and Spark backends
+            # Spark and MySQL backends
             lambda t, win: (t.double_col == 0).notall().over(win),
             lambda t: (
                 t.double_col.expanding()
@@ -143,7 +146,9 @@ from ibis.tests.backends import (
                 .astype(bool)
             ),
             id='cumnotall',
-            marks=pytest.mark.xfail_backends((Impala, PostgreSQL, Spark)),
+            marks=pytest.mark.xfail_backends(
+                (Impala, PostgreSQL, Spark, MySQL)
+            ),
         ),
         param(
             lambda t, win: t.double_col.sum().over(win),

--- a/ibis/tests/backends.py
+++ b/ibis/tests/backends.py
@@ -418,10 +418,21 @@ class MySQL(Backend, RoundHalfToEven):
         # mariadb supports window operations after version 10.2
         # but the sqlalchemy version string looks like:
         # 5.5.5.10.2.12.MariaDB.10.2.12+maria~jessie
+        # or 10.4.12.MariaDB.1:10.4.12+maria~bionic
+        # example of possible results:
+        # https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_3/
+        # test/dialect/mysql/test_dialect.py#L244-L268
         con = self.connection
         if 'MariaDB' in str(con.version):
             # we might move this parsing step to the mysql client
-            version = tuple(map(int, str(con.version).split('.')[7:9]))
+            version_detail = con.con.dialect._parse_server_version(
+                str(con.version)
+            )
+            version = (
+                version_detail[:3]
+                if version_detail[3] == 'MariaDB'
+                else version_detail[3:6]
+            )
             self.__class__.supports_window_operations = version >= (10, 2)
         elif con.version >= parse_version('8.0'):
             # mysql supports window operations after version 8

